### PR TITLE
Tests for updating/deleting devices

### PR DESCRIPTION
**Pull request by richvdh**
(This includes a test which will currently fail, but is fixed by
https://github.com/matrix-org/synapse/pull/950)

### CODE REVIEW

**Comment by richvdh** [comment]
matrixbot: please retest

**Comment by richvdh** [comment]
retest this please, even

File: tests/10apidoc/12device_management.pl [hunk 1]
```diff
@@ -94,3 +108,107 @@ sub matrix_get_device {
...
+            method  => "GET",
+            uri     => "/r0/sync",
+         );
+      })->then( sub {
+         # delete the device
+         do_request_json_for(
+            $user,
+            method => "DELETE",
+            uri    => "/unstable/devices/${DEVICE_ID}",
+           );
```
Comment (leonerd) [inline tests/10apidoc/12device_management.pl]: Indentation of this line

File: tests/10apidoc/12device_management.pl [hunk 2]
```diff
@@ -94,3 +108,107 @@ sub matrix_get_device {
...
+         )->main::expect_http_401;
+      })->then( sub {
+         # so should our refresh token
+         $user->http->do_request_json(
+            method => "POST",
+            uri    => "/r0/tokenrefresh",
+            content => {
+               refresh_token => $other_login->refresh_token,
+            },
+          )->main::expect_http_403;
```
Comment (leonerd) [inline tests/10apidoc/12device_management.pl]: Indentation looks wrong here too

File: tests/10apidoc/12device_management.pl [hunk 3]
```diff
@@ -94,3 +108,107 @@ sub matrix_get_device {
...
+      my ( $user ) = @_;
+
+      do_request_json_for(
+         $user,
+         method => "PUT",
+         uri    => "/unstable/devices/unknown_device",
+         content => {
+            display_name => "new display name",
+         },
+        )->main::expect_http_404;
```
Comment (leonerd) [inline tests/10apidoc/12device_management.pl]: Indentation seems to have gone wrong here

File: tests/10apidoc/12device_management.pl [hunk 4]
```diff
@@ -37,6 +37,20 @@ sub matrix_get_device {
...
+   requires => [ local_user_fixture( with_events => 0 ) ],
+
+   do => sub {
+      my ( $user ) = @_;
+
+      do_request_json_for(
+         $user,
+         method => "GET",
+         uri    => "/unstable/devices/unknown_device",
+        )->main::expect_http_404;
```
Comment (leonerd) [inline tests/10apidoc/12device_management.pl]: Indentation

**Comment by leonerd** [comment]
LGTM aside from some tiny indentation misalignments

**Pull request merged**